frontend/src/features/admin-form/responses/ResponsesPage/storage/utils/EncryptedResponseCsvGenerator/EncryptedResponseCsvGenerator.ts

```diff
@@ -14,6 +14,8 @@ type UnprocessedRecord = Merge<
   { record: Dictionary<Response> }
 >
 
+export const PURE_NUMBER_REGEXP = new RegExp('^[+-][0-9]+')
+export const FORMULA_INJECTION_REGEXP = new RegExp('^[+=@-].*')
```

Comment:
Minor comments:
* No need to use the `RegExp` constructor, unless we are constructing a pattern from parts
* Pure numbers regexp should have an end anchor as well. Otherwise in its current form, it still allows formulas like this: `+3*SUM(A1)` (in fact do add a test for an input like that please)
* Checking for `.*` at the end of a regexp can be discarded

```suggestion
export const PURE_NUMBER_REGEXP = /^[+-]\d+(\.\d+)?$/
export const FORMULA_INJECTION_REGEXP = /^[+=@-]/
```

Maybe we can simplify `PURE_NUMBER_REGEXP` to `/^[+-][\d.]+$/` ? That's not exact number regexp, but it's safe from an injection perspective.

Reply:
thinking about whether we should check just `/^[+-.\d]*$/` for expressions like `-4+2-0` or something like that, since i guess those are formulas but i don't think they actually pose any risk (correct me if i'm wrong). It might be simpler to describe to admins too, to explain the behavior of what we add quotes to and what we don't

Reply:
thinking about whether we should check just `/^[+-.\d]*$/` for expressions like "-4+2-0" or something like that, since i guess those are formulas but i don’t think they actually pose any risk (correct me if i’m wrong). It might be simpler to describe to admins too, to explain the behavior of what we add quotes to and what we don’t

Reply:
@justynoh yeah I don't think they pose any risks if they are just digits and + or -. Didn't come across that being mentioned in any of the sites which talked about formula injection - the malicious commands all contained at least 1 alphabet. 
Made the change in [858f4dc](https://github.com/opengovsg/FormSG/pull/4950/commits/858f4dc713acccb1b2d39af79f4b1db366b39d41), decided to go with `/^[+-.\d]+$/`

Reply:
@timotheeg thanks tim! added the additional test here [c0b1511](https://github.com/opengovsg/FormSG/pull/4950/commits/c0b15116ee5752bcee930794fbe940261b52ff71)

Reply:
@timotheeg 's offline comment: `-` is a range separator, so should only be added at the end of the regexp. otherwise it captures ASCII characters between `+` and `.`

Reply:
out of curiosity, then, if they all contain at least 1 alphabet, why not just use a single regex `/^[-+@=].*[a-zA-Z]/` haha. Maybe there is a case I'm not seeing

Reply:
@justynoh that regex wouldn't capture other symbols (e.g. `|`) and formulas like `=SUM(A1)`. also I think it's better to play it safe and not have Excel react to any formulas at all (other than ones with numbers and + | -, which we know is harmless).